### PR TITLE
`oauth`: slack dual flow connectors + gated setup

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Add dual support for old and new connection ids for Slack
- Add gated use of `oauth` to setup slack in front

## Risk

Low (gated)

## Deploy Plan

- update necessary secrets
- deploy `oauth` (to pull new slack code)
- deploy `connectors`
- deploy `front`

### CODE REVIEW

File: front/pages/w/[wId]/builder/data-sources/managed.tsx [hunk 1]
```diff
@@ -112,9 +112,11 @@ export async function setupConnection({
 
   if (
     isOAuthProvider(provider) &&
-    // Behind flag oauth-ready providers
-    // owner.flags.includes("test_oauth_setup") &&
-    ["github"].includes(provider)
+    // `oauth`-ready providers
+    (["github"].includes(provider) ||
```
Comment (fontanierh) [inline front/pages/w/[wId]/builder/data-sources/managed.tsx:116]: maybe just have one single array.includes ?

**Review by fontanierh — COMMENTED** [review]
just a nit but lgtm

**Review by fontanierh — APPROVED** [review]

File: front/pages/w/[wId]/builder/data-sources/managed.tsx (hunk 1 above)
Comment (spolu) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx:116]: Hmm these are not the same. One is the gated ones, the other the live ones

**Comment by spolu** [comment]
Tested locally

File: front/pages/w/[wId]/builder/data-sources/managed.tsx (hunk 1 above)
Comment (fontanierh) [reply front/pages/w/[wId]/builder/data-sources/managed.tsx:112]: mb indeed

**Pull request merged**